frb_codegen/src/method_utils.rs

```diff
@@ -0,0 +1,181 @@
+use crate::ir::IrType::{self, StructRef};
+use crate::ir::{IrField, IrTypeStructRef};
+use crate::ir::{IrFile, IrFunc};
+use convert_case::{Case, Casing};
+
+const STATIC_METHOD_MARKER: &str = "__static_method___";
+const METHOD_MARKER: &str = "__method___";
+
+pub(crate) struct MethodNamingUtil;
+
+impl MethodNamingUtil {
+    // Is the function name for a non static method?
+    pub fn is_non_static_method(s: &str) -> bool {
+        s.contains(METHOD_MARKER)
+    }
+
+    // Is the function name for a static method?
+    pub fn is_static_method(s: &str) -> bool {
+        s.contains(STATIC_METHOD_MARKER)
+    }
+
+    // Tests if the function is `f` is a static method for struct with name `struct_name`
+    pub fn is_static_method_for_struct(f: &&IrFunc, struct_name: &str) -> bool {
```

Comment:
same "non-pub" comments applies to most functions in this file.

e.g. whoever uses is_static_method_for_struct can change to deserialize + check struct_name equality